src/Maestro/Maestro.Web/Api/v2019_01_16/Controllers/GoalController.cs

```diff
@@ -0,0 +1,93 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Threading.Tasks;
+using Maestro.Data;
+using Maestro.Data.Models;
+using Maestro.Web.Api.v2019_01_16.Models;
+using Microsoft.AspNetCore.ApiVersioning;
+using Microsoft.AspNetCore.ApiVersioning.Swashbuckle;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.KeyVault;
+using Microsoft.EntityFrameworkCore;
+using System.Net;
+
+namespace Maestro.Web.Api.v2019_01_16.Controllers
+{
+    [Route("goals")]
+    [ApiVersion("2019-01-16")]
+    public class GoalController : ControllerBase
+    {
+        protected readonly BuildAssetRegistryContext _context;
+        public GoalController(BuildAssetRegistryContext context)
+        {
+            _context = context;
+        }
+
+        /// <summary>
+        /// Sets a build time in minutes <see cref="Goal"/> for a given Definition in a Channel. This is captured for the Power BI Dashboard -Internal Report under .Net Core Engineering Services workspace.
+        /// </summary>
+        /// <param name="goalData">An object containing build time goal in minutes <see cref="Goal"/></param>
+        /// <param name="channelName">Channel Name for the build time Eg. .Net Core 5</param>
+        /// <param name="definitionId">Azure DevOps pipeline Definition Id</param>
+        [HttpPut]
+        [SwaggerApiResponse(System.Net.HttpStatusCode.OK, Type = typeof(Models.Goal), Description = "Sets a build time goal (in minutes) for a given Definition in a Channel.")]
+        [ValidateModelState]
+        public virtual async Task<IActionResult> Create([FromBody, Required] Goal.GoalData goalData,[Required] String channelName , [Required] int definitionId)
+        {
+            Data.Models.Channel channel = await _context.Channels
+                .Where(c => c.Name.Equals(channelName)).FirstOrDefaultAsync();
+            if (channel == null)
+            {
+                return NotFound();
+            }
+            Data.Models.GoalTime goal = await _context.GoalTime
+                .Where(g => g.DefinitionId == definitionId && g.ChannelId == channel.Id).FirstOrDefaultAsync();
+            if (goal == null)
+            {
+                goal = new Data.Models.GoalTime
+                {
+                    DefinitionId = definitionId,
+                    Minutes = goalData.Minutes,
+                    ChannelId = channel.Id
+                };
+                await _context.GoalTime.AddAsync(goal);
+            }
+            else
+            // If the combination of Channel and DefinitionId already exists then update the exisiting record
```

Comment:
A personal preference, but I don't like comments between the if/else and the opening {, it makes my brain have to work harder to figure out where the block starts and what it goes with.  I'd move it down a line so its `else { //comment`

Reply:
okay will remove it :)